appcues/src/main/java/com/appcues/statemachine/Action.kt

```diff
@@ -1,16 +1,19 @@
 package com.appcues.statemachine
 
 import com.appcues.data.model.Experience
+import com.appcues.data.model.StepReference
 
 internal sealed class Action {
     data class StartExperience(val experience: Experience) : Action()
-    data class StartStep(val stepReference: StepReference) : Action()
-    object RenderStep : Action()
+    data class MoveToStep(val stepReference: StepReference) : Action()
```

Comment:
added this MoveToStep action to Separate concern from StartStep, which was being used by both RenderingStepState and EndingStepState.

now RenderingStepState takes MoveToStep and decides if it should go to EndingStepState(StartStep) this way we can pass in more valid information thru StartStep like nextFlatStepIndex and nextStepContainerIndex, both of which needs to be valid when we get there. this way we
 1. avoid getting locked into EndingStepState with possibly an invalid container step (which should never happen)
 2. if any error when getting those values, we can keep the current RenderingStepState and log the error, meaning user will continue to see the experience and can find another way to get out of it (which should also never happen but we safe check)

Reply:
I drafted what I think it would look like now.

https://whimsical.com/mobile-experience-states-Cb9WqPWhfUjb7tin4t5LYf

the MoveToStep(stepReference) is useful because it saves us the need to re-check the nextIndex and container, currently we have basically duplicated code in both transitions, and we have noted in commends that from EndingStep to BeginningStep some of those checks are safe guards that SHOULD NEVER happen.

Reply:
thanks for the updated diagram, that helps me understand - feels like an implementation detail we can live with being slightly different on Android then, in exchange for the code simplification. The overall state flow diagram is still generally matching, which is the key thing I think.